Minor docstring updates

alfonsoSR opened the pull request:
- Imported exposure of InterpolationSettings in expose_data.cpp to remove associated C++ objects from docstrings
- Added minimal docstring to VehicleSystems.set_default_transponder_turnaround_ratio_function
- Imported exposure of ObservationCollection in expose_observations_wrapper.cpp (same as above)
- Adjusted build order in expose_estimation remove C++ objects from docstrings
- Exposed constructor of InterpolatorGeneratorSettings to be able to initialize them directly from the object
- Updated build.py to display `Stubs generated successfully` at the end. Requested by @luigigisolfi  to make it clear that stubgen errors don't matter

alfonsoSR committed:
Rearranged code to fix docstring of create_ephemeris function

alfonsoSR committed:
Some docstring updates

- Imported exposure of InterpolationSettings in expose_data.cpp to
  remove associated C++ objects from docstrings
- Added minimal docstring to
  VehicleSystems.set_default_transponder_turnaround_ratio_function
- Imported exposure of ObservationCollection in
  expose_observations_wrapper.cpp (same as above)
- Adjusted build order in expose_estimation remove C++ objects from
  docstrings
- Exposed constructor of InterpolatorGeneratorSettings to be able to
  initialize them directly from the object
- Updated build.py to display `Stubs generated successfully` at the end.
  Requested by Luigi to make it clear that stubgen errors don't matter

alfonsoSR committed:
Exposed enumeration of ephemeris types

Not having it exposed would result into an error while trying to examine
ephemeris settings. I used native_enum instead of `enum_` as recommended
in the stable version of pybind11's documentation

alfonsoSR committed:
Exposed and documented function to read IFMS files

DominicDirkx commented on src/tudatpy/data/expose_data.cpp:
```diff
@@ -472,6 +472,21 @@ void expose_data( py::module &m )
+
+           Two of the columns of an IFMS file contain, respectively, the Doppler averaged frequency and a tropospheric correction for the station. When the `apply_tropospheric_correction` option is set to true, the content of the first column is modified by subtracting the values in the second.
+
+           :param file_name: String representing the path to the file to be loaded
```
this formatting looks different from what we normally use, is there a reason behind it?

DominicDirkx commented on src/tudatpy/dynamics/environment/expose_environment.cpp:
```diff
@@ -806,10 +982,7 @@ inside a `Body` instance and used in observation corrections or environmental qu
 
      )doc" )
             .def( "set_reference_point",
```
I see several of these changes, is this on purpose or auto-formatting?

alfonsoSR commented on src/tudatpy/data/expose_data.cpp:
```diff
@@ -36,7 +36,10 @@ namespace data
 
 void expose_data( py::module &m )
 {
-    // py::module_::import( "tudatpy.math.interpolators" );
```
Those import statements are used to remove C++ types from docstrings. The only reason why not having them does not produce runtime errors is that we compile the kernel into a single shared object, so they are "only" important for stubs and the API.

What I did is equivalent to replacing a star import from `tudatpy.math.interpolators` with an import of `InterpolatorSettings`, and it belongs in this PR :)

alfonsoSR committed:
Expose missing options of TrackingDataType enum

alfonsoSR committed:
Expose compute_current_frequency method of PiecewiseLinearFrequencyInterpolator

alfonsoSR committed:
Ground station and motion settings

- Exposed enumeration of ground station motion models
- Exposed constructor of GroundStationMotionSettings + model_type
  property
- Exposed BodyDeformationMotionSettings
- Exposed properties of GroundStationSettings
- Exposed functions to get the position and velocity of VLBI stations
  from glo.sit and glo.vel

alfonsoSR committed:
Expose GcrsToItrsRotationModelSettings

alfonsoSR committed:
Exposed constructor of SingleObservationSet

alfonsoSR committed:
Exposure of ODF interfaces

- Exposed classes to interact with the OdfRawFileContents from python
- Created an enumeration of data types in the ODF files, and refactored
  code to make it compatible with it
- Some commit noise due to automatic formatting (To discuss in next
  developer meeting)

alfonsoSR committed:
Exposed some properties of StationFrequencyInterpolator

alfonsoSR committed:
Resolve merge conflicts - develop_10

alfonsoSR committed:
Remove unused counter from expose_environment.cpp - Compilation warning

alfonsoSR committed:
Turned TroposphericMappingModel into enum class

alfonsoSR committed:
Fixed deprecated imports + removed external imports from init in mission_data_downloader

alfonsoSR committed:
Exposure of TroposphericMappingModel as native enum

alfonsoSR committed:
Update clang-format action to run on develop_10

alfonsoSR commented:
I run the C++ tests locally, because I modified one and it made the CI fail. That test is passing now, but I get 12 failing. I don't think any of them is related to code I have touched. Is this expected?

```
The following tests FAILED:
         63 - test_ephemerides_SynchronousRotationalEphemeris (Failed)
         78 - test_gravitation_GravityFieldVariations (Failed)
        119 - test_observation_models_DopplerMeasuredFrequencyObservationModel (Failed)
        136 - test_orbit_determination_TidalPropertyEstimation (Failed)
        146 - test_orbit_determination_NonSequentialStateEstimation (Failed)
        148 - test_orbit_determination_ConsiderParameters (Failed)
        186 - test_observation_partials_ObservedFrequencyPartials (Failed)
        208 - test_propagators_NonSequentialPropagation (Failed)
        215 - test_propagators_HybridArcVariationalEquations (Failed)
        216 - test_propagators_HybridArcStateTransitionMatrixInterface (Failed)
        219 - test_propagators_NonSequentialVariationalEquations (Failed)
        276 - test_simulation_EnvironmentModelSetup (Failed)
Errors while running CTest
```

alfonsoSR committed:
Remove unused counter from expose_environment + fixed modified C++ test

alfonsoSR committed:
Removed unused counter from expose_environment

DominicDirkx commented:
The CI seems to pass fine. Could you post the full test log file with the test output?

DominicDirkx committed:
Merge branch 'develop_10' into feature/minor-docstring-updates

DominicDirkx committed:
Modified warnings; added int constructor for Time

DominicDirkx committed:
Merge

DominicDirkx commented:
The CI passes, and the test you mentioned compile locally. So I'm merging this :)
